kani-compiler/src/codegen_cprover_gotoc/archive.rs

```diff
@@ -2,114 +2,42 @@
 //
 // Modifications Copyright Kani Contributors
 // See GitHub history for details.
-// This file is a lightly modified version of upstream rustc:
+// This file is a heavily modified version of upstream rustc:
 //     compiler/rustc_codegen_cranelift/src/archive.rs
 
-// Along with lifting the deps:
-//  object = { version = "0.27.0", default-features = false, features = ["std", "read_core", "write", "archive", "coff", "elf", "macho", "pe"] }
-//  ar = "0.8.0"
-
-// Also: I removed all mentions of 'ranlib' which caused issues on macos.
-// We don't actually need these to be passed to a real linker anyway.
-// 'ranlib' is about building a global symbol table out of all the object
-// files in the archive, and we just don't put object files in our archives.
-
 //! Creation of ar archives like for the lib and staticlib crate type
+//! We now call the ArchiveBuilder directly, so we don't bother trying to fit into the rustc's
+//! `ArchiveBuilder`.
 
+use rustc_session::Session;
 use std::fs::File;
-use std::io::{self, Read, Seek};
 use std::path::{Path, PathBuf};
 
-use rustc_codegen_ssa::back::archive::{ArchiveBuilder, ArchiveBuilderBuilder};
-use rustc_session::Session;
-
-use object::read::archive::ArchiveFile;
-use object::ReadCache;
-
-#[derive(Debug)]
-enum ArchiveEntry {
-    FromArchive { archive_index: usize, file_range: (u64, u64) },
-    File(PathBuf),
-}
-
-pub(crate) struct ArArchiveBuilder<'a> {
+pub(crate) struct ArchiveBuilder<'a> {
```

Comment:
We don't have access to this yet, since we are still using an older version of `rustc`. But once we do update to a newer version, I agree that we should be able to completely remove this file and just use `ArArchiveBuilder` directly.